### ISSUE TITLE
Bitbucket cloud support

### ISSUE THREAD

**Issue by SemenovAlexander** [jenkinsci/violation-comments-to-stash-plugin#31]
Hi, does this plugin support work with bitbucket cloud, atlassian hosted repository?

I've tried various settings, but always see http error in logs:
`{"type": "error", "error": {"message": "Resource not found", "detail": "There is no API hosted at this URL.\n\nFor information about our API's, please refer to the documentation at: https://developer.atlassian.com/bitbucket/api/2/reference/"}}`

**Comment by tomasbjerre** [comment]
It does not. Its a completely different product.

**Comment by LuisPiedra** [comment]
It would be really nice to have this for Bit Bucket Cloud also. I see most of the work would be creating a Violations Lib implementation. I've found some stalled development for this implementation, but I'm not sure about the status or completeness of the project.
https://github.com/ghusse/violations-comments-to-bitbucket-cloud-lib

I think I'll give it a try. Do you think it would be better to add support in this plugin, or maybe it would be better to create a new plugin for cloud support?

**Comment by tomasbjerre** [comment]
Yes:
* A new library, like the one you have found.
* A new Jenkins-plugin.

The argument for new plugin is that Bitbucket Server and Cloud are completely different products.

* Cloud implemented in Python (I think), Server implemented in Java. They will never merge.
* Completely different API:s. Completely different clients for the API:s.

**Comment by khoatd92** [comment]
Hi, so for now we can't add comments for Bitbucket Cloud's pull request, right?
Or do we have any plugins to do that?

**Comment by tomasbjerre** [comment]
I don't know any plugin that can add violation comments like this one.

**Comment by khoatd92** [comment]
I see. Hmm can we use webhook or something like that to do it? CircleCI or something like that. Because Bitbucket server is not cheap so I hope we can find a better solution for Bitbucket Cloud.

**Comment by tomasbjerre** [comment]
Triggering on a pull request in Bitbucket Cloud is not problem. There are plugins for that. If that is what you are asking when you say webhook.

I don't know of any library that can comment a pull request in Bitbucket Cloud.

It sounds like @LuisPiedra will work on it. Will you?

**Comment by tomasbjerre** [comment]
Probably worth mentioning that there are plugins for GitHub and GitLab if you want a free hosting service:

* https://github.com/jenkinsci/violation-comments-to-github-plugin
* https://github.com/jenkinsci/violation-comments-to-gitlab-plugin

**Comment by LuisPiedra** [comment]
I have submitted a pull request to complete Bitbucket Cloud lib
https://github.com/ghusse/violations-comments-to-bitbucket-cloud-lib/pull/12

I have also some work in progress for a new plugin using Cloud library, hopefully I will be able to share it soon.

**Comment by nvtkaszpir** [comment]
@LuisPiedra any progress on this?

**Comment by Kinmarui** [comment]
Wow, this may be obvious to some but I just spent whole day fighting with credentials in jenkins / bitbucket, just to arrive at this issue. Maybe it would be good idea to add this warning on main readme page?

**Comment by wozman** [comment]
Yes as @Kinmarui says this really needs to be made very clear at the top of the docs, as I also wasted the best part of a day trying to get this working with Bitbucket Cloud.

**Comment by tomasbjerre** [comment]
I have published a command line utility in NPM that does this with Cloud now:

https://github.com/tomasbjerre/violation-comments-to-bitbucket-cloud-command-line
